### PR TITLE
(IAC-137) - Litmus test framework implemented

### PR DESCRIPTION
**Pull request by david22swan**
Screen Shot 2020-04-23 at 7 25 31 PM

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        ssh-add ~/.ssh/id_rsa-acceptance
+        bundle exec bolt --modulepath spec/fixtures/modules -i ./inventory.yaml plan run satellite_pe_tools::test_01_provision
+        bundle exec bolt --modulepath spec/fixtures/modules -i ./inventory.yaml plan run satellite_pe_tools::test_02_server_setup
+    - name: Inventory
+      run: |
+        cat ./inventory.yaml
+    - name: Run tests
+      run: |
+        env
+        bundle exec bolt --modulepath spec/fixtures/modules -i ./inventory.yaml plan run satellite_pe_tools::test_03_test_run
```
Comment (sheenaajay) [inline .github/workflows/release.yml:67]: @david22swan can we use the integeration tag in rakefile and run the tests using the following command
https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/master/Rakefile#L94
https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/master/.github/workflows/weekly.yml#L59

File: Gemfile [hunk 2]
```diff
@@ -17,6 +17,8 @@ ruby_version_segments = Gem::Version.new(RUBY_VERSION.dup).segments
 minor_version = ruby_version_segments[0..1].join('.')
 
 group :development do
+  gem 'ed25519', '>= 1.2', '< 2.0'
```
Comment (sheenaajay) [inline Gemfile:20]: Should we add them to .sync.yaml file too

File: .github/workflows/release.yml (hunk 1 above)
Comment (sheenaajay) [reply .github/workflows/release.yml:67]: Could you please add weekly workflow too. Thank you.

**Comment by sheenaajay** [comment]
@david22swan Good work.

File: spec/spec_helper_acceptance_local.rb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    change_target_host(satellite)
+    expand_satellite_disk(satellite) if get_provisioner(satellite) == 'vmpooler' # && satellite['disks']
+    install_satellite(satellite)
+    reset_target_host
+    generate_and_transfer_satellite_cert_from_sat_to_pe(satellite)
+  end
+end
+
+def change_target_host(role)
+  @orig_target_host = ENV['TARGET_HOST']
```
Comment (sanfrancrisko) [inline spec/spec_helper_acceptance_local.rb:34]: I see `change_target_host` being invoked in two `RSpec.before` hooks:

- `:suite`: From [here](https://github.com/puppetlabs/puppetlabs-satellite_pe_tools/pull/129/files#diff-08070fc30be3f9dd014e20b26f80934bR25)
- `:all`: [Calling `satellite_update_setting`](https://github.com/puppetlabs/puppetlabs-satellite_pe_tools/pull/129/files#diff-4bcca2bf146303382123f641fdcb1aa1R11) -> then [here](https://github.com/puppetlabs/puppetlabs-satellite_pe_tools/pull/129/files#diff-08070fc30be3f9dd014e20b26f80934bR170)

Will this mean `@orig_target_host` is what we expect if it's being called twice before the tests execute?

Comment (david22swan) [reply spec/spec_helper_acceptance_local.rb:34]: change_target_host is only called to set the target to the satellite with reset_target_host swapping it back to the master.

File: Gemfile (hunk 2 above)
Comment (david22swan) [reply Gemfile:20]: Done

File: .github/workflows/release.yml (hunk 1 above)
Comment (david22swan) [reply .github/workflows/release.yml:67]: Tried the integration tag but it kept throwing an error, `Target 'litmus_localhost' not found in inventory.yaml`.
Think we did something different at some point.
Weekly workflow added.

Comment (david22swan) [reply .github/workflows/release.yml]: Done

File: spec/spec_helper_acceptance_local.rb (hunk 3 above)
Comment (david22swan) [reply spec/spec_helper_acceptance_local.rb]: `@orig_target_host = ENV['TARGET_HOST']` has been removed

**Pull request merged**